Setup Vagrant

adhikaribiraj opened the pull request.

Commit:
Setup Vagrant

KevinTran000 committed:
- Resolved ujson dependency issue.

KevinTran000 committed:
- Modified Vagrantfile to install other dev. tools
-  Also print statements for easier output reading.

KevinTran000 committed:
Merge branch 'master' of https://github.com/StoneyJackson/BEAR-Necessities-Market into setup-vagrant

KevinTran000 commented:
Resynced with master, merge conflict resolved.

To Do List:
- [x] Revise Documentation

KevinTran000 committed:
Slight refactor

KevinTran000 committed:
Slight refactor

StefieNguyen committed:
updated README to setup project with Vagrant

StefieNguyen committed:
Merge branch 'setup-vagrant' of https://github.com/StoneyJackson/BEAR-Necessities-Market into setup-vagrant

KevinTran000 committed:
- Readme tweaks.

- Added links to readme
- Standardized a few wordings
- Wrapped command in code backticks
- Reword a section to make more apparent that a section terminal is required.

KevinTran000 committed:
- Readme tweaks

- Clarifying a section

KevinTran000 committed:
- Readme tweaks

- Standardizing section headers wih colons.

KevinTran000 committed:
- Readme tweaks

- Fixed typo
- Fixed capitalization errors

KevinTran000 committed:
Merge branch 'master' of https://github.com/LibreFoodPantry/BEAR-Necessities-Market into setup-vagrant

KevinTran000 commented:
Resynced with master. Manual testing is required to ensure it works.

Other than that, there is no more pending work for this PR so it is now ready for review/testing.

KevinTran000 commented on README.md:
```diff
@@ -32,28 +32,45 @@ Some overview description will go here
 
 > Or take a look at our detailed [getting started][getting-started] instructions.
```
This artifact is still here pointing at the old set up instructions. Not sure if want to get rid of it or keep it there in case someone can't get vagrant on their system for whatever reason.

adhikaribiraj replied:
The duplicate getting started guide was removed.

StoneyJackson commented:
First, it looks like this is a problem on MacOS. So I found this article.

https://superuser.com/questions/1255845/how-to-setup-virtualbox-port-forwarding-on-mac-os-x-to-ubuntu-16-04

It says that the application needs to be listening on "every IP interface". And the way to do that is in this documentation.

http://flask.pocoo.org/docs/1.0/quickstart/

In short....

```
flask run --host=0.0.0.0
```

At least, worked on mine.

KevinTran000 committed:
- Readme tweaks

- Added `--host=0.0.0.0` switch to `flask run` instruction

Commit:
Remove duplicate getting started guide

Commit:
Delete the getting started file

KevinTran000 committed:
Merge branch 'master' of https://github.com/LibreFoodPantry/BEAR-Necessities-Market into setup-vagrant

ndion617 commented:
On windows having an issue with `$flask db upgrade`
`vagrant@vagrant:~$ export FLASK_APP=manage.py`
`vagrant@vagrant:~$ export SENDGRID_API_KEY="RANDOMKEY"`   
 `vagrant@vagrant:~$ export SENDGRID_DEFAULT_FROM="example@example.com"`

`vagrant@vagrant:~$ flask db upgrade`                                                      
`Usage: flask db upgrade [OPTIONS] [REVISION] `      
 `Error: Could not import "manage".`

EmiliaStormy commented:
need to update documentation to specify new terminal must be opened in BEAR-Necessities-Market

EmiliaStormy commented:
need to update documentation with how to shut down vagrant: open new terminal in BEAR-Necessities-Market enter "vagrant suspend"

EmiliaStormy commented:
add RAM requirement, was frequent freezing with 4GB RAM

KevinTran000 commented:
> 
> 
> On windows having an issue with `$flask db upgrade`
> `vagrant@vagrant:~$ export FLASK_APP=manage.py`
> `vagrant@vagrant:~$ export SENDGRID_API_KEY="RANDOMKEY"`
> `vagrant@vagrant:~$ export SENDGRID_DEFAULT_FROM="example@example.com"`
> 
> `vagrant@vagrant:~$ flask db upgrade`
> `Usage: flask db upgrade [OPTIONS] [REVISION] ` `Error: Could not import "manage".`

You missed a step, `cd BEAR-Necessities-Market` just before that. I'll clarify that section and stick all the commands into one code block instead of multiple smaller code blocks, as apparent in your comment, there is chance to accidentally skip step due to multiple small code blocks.



> 
> 
> need to update documentation to specify new terminal must be opened in BEAR-Necessities-Market

Fair, I neglected to specify since my local new terminal always open up a new terminal in same project directory automatically so the thought did not cross my mind. I will correct that in my next push up.



> 
> 
> need to update documentation with how to shut down vagrant: open new terminal in BEAR-Necessities-Market enter "vagrant suspend"

Good point, @StoneyJackson mentioned that to me the other day and I forgot to make a note so that I would remember. This reply will serve as my to-do list later. Thank you.

> 
> 
> add RAM requirement, was frequent freezing with 4GB RAM

Ah yes, fair. I was a bit worried how it would fair on more limited resources and looks like that confirms it. Will add section about minimum requirements in my next push.

Thank you @ndion617 and @EmiliaStormy for the review. Once the below `To Do` list is completed, I shall post another comment saying it is ready for review/testing again.

--------------------
To Do (Readme tweaks):

- [x] Clarify instruction (compress code blocks)
- [x] Clarify that new terminal should be opened in the project folder since the vagrant command needs to be executed in the same directory as the `Vagrantfile`
- [x] Add system requirements section
- [x] Add instructions for how to shutdown Vagrant (VM)

KevinTran000 committed:
- Readme Tweaks

- Added Minimum System Requirements
- Condensed some instructions for clarity
- Clarified terminal location
- Added instructions for how to shutdown, suspend, and boot up the vagrant box.

KevinTran000 commented:
I pushed up changes as suggested by the reviewers. This PR is once again ready for additional reviews/testing.

ndion617 commented:
Okay pulled it down and tested, I approve.
